### PR TITLE
Add a per-link attribute for disabling intralink validation

### LINKED ISSUE #1387: Add a per-link attribute for disabling intralink validation
**Is your request related to a problem?**

Untitled

The user guide has some dummy links used in the navbar example code snippet.

**Describe the solution you'd like**

In line with the `no-convert` attribute for `.md / .mbd` conversion to `.html`, let's add a `no-intralink-validation` (can be shorter) attribute of sorts to disable intralink validation for such cases

**Describe alternatives you've considered**
na

**Additional context**
na

### PR DESCRIPTION
**Pull request by bchenghi**
**What is the purpose of this pull request?**

- [x] Documentation update
- [ ] Bug fix
- [x] Feature addition or enhancement
- [ ] Code maintenance
- [ ] Others, please explain:

**Overview of changes:**
Fixes #1387. Related #1418. Add no-validation attribute to link and the link will not be validated. A test was added in linkProcessor.test.js

**Anything you'd like to highlight / discuss:**
Code quality and implementation.

**Testing instructions:**
npm run test

**Proposed commit message: (wrap lines at 72 characters)**
linkProcessor: add attribute to disable intralink validation

There are no known methods to disable intralink validation per-link.

There were dummy links in the user guide that failed the intralink
 validation and printed warnings in the console.

Let's add a no-validation attribute to disable intralink
 validation for such cases in the future.

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [x] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No blatantly unrelated changes
- [x] Pinged someone for a review!

### CODE REVIEW

**Comment by damithc** [comment]
Quick question: Would this work for Markdown-style links e.g., `[link](http://www.markbind.com)`?

**Comment by bchenghi** [comment]
@damithc Yes Prof. It can be done by adding the attribute inline , e.g. `Click [here](http://www.markbind.com){no-intralink-validation}`.

**Comment by damithc** [comment]
> @damithc Yes Prof. It can be done by adding the attribute inline , e.g. `Click [here](http://www.markbind.com){no-intralink-validation}`.

Good. Can we make the attribute name shorter? e.g., `no-validation`

**Comment by bchenghi** [comment]
@damithc Thank you Prof, I have renamed the attribute.

File: packages/core/src/html/linkProcessor.js [hunk 1]
```diff
@@ -125,14 +125,22 @@ function isValidFileAsset(resourcePath, config) {
 
 /**
  * Serves as an internal intra-link validator. Checks if the intra-links are valid.
- * If the intra-links are not suspected to not be valid, a warning message will be logged.
+ * If the intra-links are suspected to not be valid and they do not have the no-validation
```
Comment (ang-zeyu) [inline packages/core/src/html/linkProcessor.js:128]:
```suggestion
 * If the intra-links are suspected to be invalid and they do not have the no-validation
```

**Review by ang-zeyu — COMMENTED** [review]
Needs documentation updates, I think [here](https://markbind.org/userGuide/formattingContents.html#links) (or anywhere you think would suit better 🙂)

And one tiny nit:

**Comment by bchenghi** [comment]
Thank you @ang-zeyu, I have updated the JSDoc and the links section in the user guide.

**Review by ang-zeyu — APPROVED** [review]
Nice work, thanks! @bchenghi

**Pull request merged**